src/ctia/http/server.clj

```diff
@@ -122,6 +122,17 @@
         (update :headers (fn [response-headers]
                            (into headers response-headers)))))))
 
+(defn wrap-txt-accept-header
+  "Enforces the `accept` request header to `text/plain` when the
+   uri ends with `.txt`. Mainly used by the `/feed/{id}/view.txt` endpoint"
+  [handler]
+  (fn [{:keys [uri] :as request}]
+    (let [new-request
+          (cond-> request
+            (string/ends-with? uri ".txt")
```

Comment:
Probably not all endpoints support .txt, but this middleware would suggest so.

i.e. a client shouldn't be able to dictate what a server accepts? (or at least not unilaterally)

(perhaps I'd try to code an integration test showing why this is needed)

Reply:
The `.txt` suffix is only supported if the route is defined, otherwise a 404 is returned.

This middleware is here to set default mime type accepted by the client if the `.txt` extension is used at the end of the request URI. If not set, the response encoder for json is used by default. It comes from the `wrap-restful-response` middleware that takes the first format defined in its options (https://github.com/metosin/compojure-api/blob/8b5122ad308f3ed17c452bbfc3cb4e3ea2cae84c/src/compojure/api/middleware.clj#L154).

The integration test is here https://github.com/threatgrid/ctia/pull/1074/files?diff=split&w=1#diff-c0821316e41b318d4f743b7391b511c7ab557d55f3b61dc643b1ca1767cd1949R159. I forgot to push the tests :)

Reply:
> The .txt suffix is only supported if the route is defined, otherwise a 404 is returned.

Good point. Probably I was thinking of Rails where clients can add .json/.txt/.xml/... to _any_ route and will affect content negotiation.